### ISSUE TITLE
test_participant_selection fails when using the tester backend

### ISSUE THREAD

**Issue by pcppcp** [raiden-network/raiden#1865]
As produced by `tests/integration/long_running/flaky/test_token_networks.py`

```
Traceback (most recent call last):
  File "src/gevent/greenlet.py", line 716, in gevent._greenlet.Greenlet.run
  File "/home/xoza/src/raiden/raiden/tasks.py", line 56, in _run
    self.poll_for_new_block()
  File "/home/xoza/src/raiden/raiden/tasks.py", line 95, in poll_for_new_block
    result = callback(current_block, chain_id)
  File "/home/xoza/src/raiden/raiden/raiden_service.py", line 392, in _callback_new_block
    self.handle_state_change(state_change, current_block_number)
  File "/home/xoza/src/raiden/raiden/raiden_service.py", line 340, in handle_state_change
    on_raiden_event(self, event)
  File "/home/xoza/src/raiden/raiden/raiden_event_handler.py", line 352, in on_raiden_event
    handle_contract_send_channelsettle(raiden, event)
  File "/home/xoza/src/raiden/raiden/raiden_event_handler.py", line 311, in handle_contract_send_channelsettle
    second_locksroot,
  File "/home/xoza/src/raiden/raiden/network/proxies/payment_channel.py", line 169, in settle
    partner_locksroot=partner_locksroot,
  File "/home/xoza/src/raiden/raiden/network/proxies/token_network.py", line 786, in settle
    raise TransactionThrew('Settle', receipt_or_none)
raiden.exceptions.TransactionThrew: Settle transaction threw. Receipt=AttributeDict({'transactionHash': HexBytes('0x07763bfc6e8538356eaa21f581898b3dd01b05df553a56ef8f5cd363a6aacb73'), 'transactionIndex': 0, 'blockNumber': 125, 'blockHash': HexBytes('0xd4a68dd33499c58119708b1d07496dfe5845ff0cc2cf5d5345083ed2d8fee788'), 'cumulativeGasUsed': 30228, 'gasUsed': 30228, 'contractAddress': None, 'logs': [], 'status': 0})
2018-07-18T08:24:08Z <AlarmTask "Greenlet-0" at 0x7f8ecb13f248: _run> failed with TransactionThrew
```

Channel is closed at block 117 and settle is called at block 125. Settle timeout should be 6 blocks.

**Comment by pcppcp** [comment]
I am closing this as it happens only with tester in a specific test. No problem with geth.

**Comment by hackaugusto** [comment]
@pcppcp but was the tester implementation fixed? or is this test flaky?

**Comment by pcppcp** [comment]
The test is skipped for tester, with geth it should pass.
But the problem is probably related to #1910 where duplicated events are returned for different filters. I'll check once we address them and eventually unskip the test again.

**Comment by andrevmatos** [comment]
I fixed #1910 as part of #1869, the error caused by the duplicated events is solved in it, but this `raiden.exceptions.TransactionThrew: Settle transaction threw` still happens half ot the times. Not only with tester, but on travis with geth too. Related to #1924

**Comment by hackaugusto** [comment]
we dropped support for tester a while ago.
